server/spec/hypervisor_check_in_spec.rb

```diff
@@ -288,6 +288,20 @@ def check_hypervisor_consumer(consumer, expected_host_uuid, expected_guest_ids)
     end.should raise_exception(RestClient::BadRequest)
   end
 
+  # simple end to end test
+  it 'should allow virt-who to report hypervisor data' do
+    virtwho = create_virtwho_client(@user)
+    json ="{\"hypervisors\":" +
+          "[{" +
+          "\"name\" : \"hypervisor_999\"," +
+          "\"hypervisorId\" : {\"hypervisorId\":\"uuid_999\"}," +
+          "\"guestIds\" : [{\"guestId\" : \"guestId_1_999\"}]" +
+          "}]}";
+    job_detail = JSON.parse(virtwho.hypervisor_update(@owner['key'], json))
+    job_detail['state'].should == 'CREATED'
+    job_detail['id'].should start_with('hypervisor_update_')
```

Comment:
Admittedly I know I've been complaining about too much spec testing here, but in this case we've added a new API call that's the way forward, and left the bulk of the testing using the old one. As this stands we have a bunch of tests using the wrong API. I would look to abstract the update method used in the rspec tests, have one class using the old API, and one using the new, but then *share* the tests, no copy paste. Test coverage of the actual virt update functionality is pretty important here.